src/ttsim/plot/dag/tt.py

```diff
@@ -202,23 +258,13 @@ def _get_tt_dag_with_node_metadata(
     return selected_dag
 
 
-def convert_all_nodes_to_callables(
-    env: SpecEnvWithoutTreeLogicAndWithDerivedFunctions,
-) -> SpecEnvWithoutTreeLogicAndWithDerivedFunctions:
-    return {
-        qn: dummy_callable(obj=n, leaf_name=dt.tree_path_from_qname(qn)[-1])
-        if not callable(n)
-        else n
-        for qn, n in env.items()
-    }
-
-
 def _get_node_descriptions(
-    env: SpecEnvWithoutTreeLogicAndWithDerivedFunctions,
+    without_tree_logic_and_with_derived_functions: SpecEnvWithoutTreeLogicAndWithDerivedFunctions,  # noqa: E501
 ) -> dict[str, str]:
     """Get the descriptions of the nodes in the environment."""
+    qn_env = dt.flatten_to_qnames(without_tree_logic_and_with_derived_functions)
     out = {}
-    for qn, n in env.items():
+    for p, n in qn_env.items():
```

Comment:
Why change the running variable? I'd go for `qn, node` actually -- ended up being a sufficiently long loop.